### DISCUSSION TITLE
on_train_epoch_end() runs int he middle of epochs

### DISCUSSION THREAD

**Discussion by inigoval** [Lightning-AI/pytorch-lightning#7375] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hi
I'm trying to calculate some metrics and generate some images to save at the end of each **epoch**.
I put this code in the on_train_epoch_end() (I also tried using a custom callback) but the function seems to be called in the middle of the epochs, approximately 3-4 times per epoch. 

Surely this isn't intended behaviour? Could it be to do with me using a combined loader?

**Comment by edgarriba** [comment, marked as answer]
@inigoval thanks for reporting this. Could you provide a snippet code so that we can reproduce ?

**Reply by inigoval** [reply]
Hi Edgar,
I think that the problem was in the logger that I was using, rather than pytorch lightning and I have now fixed it.
Apologies for reporthing this before investigating it further!

**Reply by edgarriba** [reply]
@inigoval glad to hear we are bug-free here :)
